### ISSUE TITLE
NullPointerException when projecting on a class with an orphaned link

### ISSUE THREAD

**Issue by mmacfadden** [orientechnologies/orientdb#8574]
## Expected behavior  
It is possible to create orphaned links in OrientDB by creating a link from one class to another, and then deleting the document the link points too. This should not break the entire class.

## Actual behavior  
An orphan link essentially creates SELECTING on the entire class, causing a NullPointerException.

## Steps to reproduce  
This took some doing to detect what was going on and boil it down to something easily reproducible. First run this script to create the classes and add some data.

```SQL
CREATE CLASS Target;
CREATE PROPERTY Target.id String (MANDATORY TRUE, NOTNULL TRUE);

CREATE CLASS Source;
CREATE PROPERTY Source.id String (MANDATORY TRUE, NOTNULL TRUE);
CREATE PROPERTY Source.target LINK TARGET (MANDATORY TRUE, NOTNULL TRUE);

INSERT INTO TARGET SET id = "target1";

INSERT INTO SOURCE SET id = "source1", target = (SELECT FROM Target WHERE id = "target1");
```

Selecting from the Source class works fine:

```SQL
SELECT FROM Source;
```

A projection also works:

```SQL
SELECT target.id FROM Source;
```

Now delete the linked document:

```SQL
DELETE FROM Target;
```
Selecting from the source table works.
```SQL
SELECT FROM Source;
```

However, trying a projection that includes crossing the orphaned link results in a NullPointerException.

```SQL
SELECT target.id FROM Source;
```

The following stack trace appears in the console:

```
Internal server error:
java.lang.NullPointerException
	at com.orientechnologies.orient.core.sql.parser.OSuffixIdentifier.execute(OSuffixIdentifier.java:76)
	at com.orientechnologies.orient.core.sql.parser.OSuffixIdentifier.execute(OSuffixIdentifier.java:200)
	at com.orientechnologies.orient.core.sql.parser.OModifier.execute(OModifier.java:101)
	at com.orientechnologies.orient.core.sql.parser.OBaseExpression.execute(OBaseExpression.java:127)
	at com.orientechnologies.orient.core.sql.parser.OExpression.execute(OExpression.java:110)
	at com.orientechnologies.orient.core.sql.parser.OProjectionItem.execute(OProjectionItem.java:140)
	at com.orientechnologies.orient.core.sql.parser.OProjection.calculateSingle(OProjection.java:114)
	at com.orientechnologies.orient.core.sql.executor.ProjectionCalculationStep.calculateProjections(ProjectionCalculationStep.java:66)
	at com.orientechnologies.orient.core.sql.executor.ProjectionCalculationStep.access$000(ProjectionCalculationStep.java:13)
	at com.orientechnologies.orient.core.sql.executor.ProjectionCalculationStep$1.next(ProjectionCalculationStep.java:41)
	at com.orientechnologies.orient.core.sql.executor.OScriptExecutionPlan.doExecute(OScriptExecutionPlan.java:91)
	at com.orientechnologies.orient.core.sql.executor.OScriptExecutionPlan.fetchNext(OScriptExecutionPlan.java:49)
	at com.orientechnologies.orient.core.sql.parser.OLocalResultSet.fetchNext(OLocalResultSet.java:35)
	at com.orientechnologies.orient.core.sql.parser.OLocalResultSet.&lt;init&gt;(OLocalResultSet.java:29)
	at com.orientechnologies.orient.core.command.OSqlScriptExecutor.execute(OSqlScriptExecutor.java:55)
	at com.orientechnologies.orient.core.db.document.ODatabaseDocumentEmbedded.execute(ODatabaseDocumentEmbedded.java:587)
	at com.orientechnologies.orient.server.network.protocol.http.command.post.OServerCommandPostBatch.execute(OServerCommandPostBatch.java:203)
	at com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.service(ONetworkProtocolHttpAbstract.java:176)
	at com.orientechnologies.orient.server.network.protocol.http.ONetworkProtocolHttpAbstract.execute(ONetworkProtocolHttpAbstract.java:655)
	at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:69)
```

**Comment by luigidellaquila** [comment]
Hi @mmacfadden 

Thank you for reporting, I'll check it asap and let you know

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Hi @mmacfadden 

I just pushed a fix, it will be released with v 3.0.9

Thanks

Luigi

**Comment by mmacfadden** [comment]
Thanks for the great community support as always!
